lib/dap/filter/gquic.rb

```diff
@@ -23,8 +25,13 @@ def decode(data)
               while i < data.length 
                  version = data[i..i+4-1]
                  # Versions start with the letter Q
-                 if data[i] == 'Q'
-                    versions.push(version)
+                 if version[0] == 'Q'
+                    begin
+                        # try to convert parse number from string
+                        Integer(version[1..3], 10)
```

Comment:
`version[1..3].to_i` might be more readable and Ruby commonplace.  Using failure conditions to direct control flow I believe is also not recommended.  Instead maybe you could add to the conditional line above and use a Regex?  It'd match for Q followed by four digits...

Reply:
cool! changed to ```if version =~ /^Q\d{3}$/```